vault/logical_system.go

```diff
@@ -1644,8 +1644,15 @@ func (b *SystemBackend) handleMount(ctx context.Context, req *logical.Request, d
 	// is enabled in dev and test servers. We don't want to pay the cost of key
 	// generation for that KV mount in all tests.
 	if config.usingOIDCDefaultKey() && logicalType != mountTypeKV {
-		if err := identityStore.lazyGenerateDefaultKey(ctx, storage); err != nil {
-			return nil, fmt.Errorf("failed to generate default key: %w", err)
+		err := identityStore.lazyGenerateDefaultKey(ctx, storage)
+		if err != nil {
```

Comment:
So we silently ignore ErrReadOnly if we're not local?

Reply:
Ah, I had assumed we'd get the auto-forwarding from an eventual write (causing ErrReadOnly) in this handler. I will change this to handle it immediately here.

Reply:
Well, the auto-forwarding requires that the error be seen by the routing code, but err is local to this scope and thus wouldn't be seen by the routing code unless we return it.

Reply:
Fixing this up to handle it here 👍

Reply:
Okay I think I've got it right in https://github.com/hashicorp/vault/pull/25265/commits/481aef4d6c6dcb26e73e2f53ea5d4df2d10c4229. Logic following the table below.

|  | **Local**  | **Not Local** |
| - | ----------- | ------------- |
| **Not ErrReadOnly**  | return err  | return err |
| **ErrReadOnly**  | log warning  | return err |